locator/tablets.cc

```diff
@@ -100,6 +100,11 @@ tablet_migration_streaming_info get_migration_streaming_info(const locator::topo
 tablet_migration_streaming_info get_migration_streaming_info(const locator::topology& topo, const tablet_info& tinfo, const tablet_transition_info& trinfo) {
     tablet_migration_streaming_info result;
     switch (trinfo.transition) {
+        case tablet_transition_kind::rf_change:
+            if (!trinfo.pending_replica) {
+                return result;
+            }
+            [[fallthrough]];
```

Comment:
This fallthrough means, that in case we're changing RF for a keyspace and no pending replica available -- go and perform migration transition. Engaged pending_replica means that we're increasing RF for a keyspace, but the `case tablet_transition_kind::migration` below will render empty read_from, won't it?

Shouldn't rf_change transition fall-through into rebuild kind instead?

Reply:
I am a bit lost here. I based this change on this comment https://github.com/scylladb/scylladb/pull/16723#issuecomment-1979357804 & https://github.com/scylladb/scylladb/pull/16723#discussion_r1469753895 and would appreciate suggestions on how to approach this.
cc @tgrabiec

Reply:
> This fallthrough means, that in case we're changing RF for a keyspace and no pending replica available -- go and perform migration transition. Engaged pending_replica means that we're increasing RF for a keyspace, but the `case tablet_transition_kind::migration` below will render empty read_from, won't it?
> 
> Shouldn't rf_change transition fall-through into rebuild kind instead?

I'm also lost on how the transition plan is supposed to be executed, in both cases. rf increase or decrease.

@xemul please be more specific.
And this PR or particular patches in it should probably marked as fixing particular issues that you opened about the expected reaction to rf change, and I don't see that either.

Reply:
@xemul is right, rf change is like rebuild not like migration. If you fall-through into "migration" case, then read_from will be empty for rf++, whereas it should contain all the nodes we rebuild from.  

> I based this change on this comment https://github.com/scylladb/scylladb/pull/16723#issuecomment-1979357804 & https://github.com/scylladb/scylladb/pull/16723#discussion_r1469753895 and would appreciate suggestions on how to approach this.

I don't think those comments conflict with what @xemul says.

Reply:
Changed to fall through `migration` + removed redundant `if` check.